src/main/groovy/com/palantir/gradle/versions/flexversioning/FlexVersion.groovy

```diff
@@ -0,0 +1,35 @@
+// Copyright 2015 Palantir Technologies
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.palantir.gradle.versions.flexversioning
+
+class FlexVersion {
+    String domain = null;
+    String commitCount = null;
+    String gitHash = null;
+    String fullVersion = null;
+
+    boolean dirty = false;
+    boolean tag = false;
+
+    public FlexVersion (String domain, String commitCount, String gitHash, boolean dirty) {
```

Comment:
this class should probably a build() method that the toString() method just calls rather than storing the fullVersion as a variable and computing it outside the object.

example use case would be if i wanted to get the flex version object, use its parts for almanac publishing, then assign the full version string as metadata. I want to do all of those steps using the same flex version object.